pyro/distributions/torch/one_hot_categorical.py

```diff
@@ -1,44 +1,33 @@
 from __future__ import absolute_import, division, print_function
 
 import torch
-from torch.autograd import Variable
 
 from pyro.distributions.one_hot_categorical import OneHotCategorical as _OneHotCategorical
 from pyro.distributions.torch_wrapper import TorchDistribution
-from pyro.distributions.util import copy_docs_from, get_clamped_probs
+from pyro.distributions.util import copy_docs_from
 
 
 @copy_docs_from(_OneHotCategorical)
 class OneHotCategorical(TorchDistribution):
     enumerable = True
 
     def __init__(self, ps=None, logits=None, *args, **kwargs):
-        ps = get_clamped_probs(ps, logits, is_multidimensional=True)
-        torch_dist = torch.distributions.Categorical(ps)  # TODO switch to OneHotCategorical
-        x_shape = ps.shape
+        torch_dist = torch.distributions.OneHotCategorical(probs=ps, logits=logits)
+        x_shape = ps.shape if ps is not None else logits.shape
         event_dim = 1
         super(OneHotCategorical, self).__init__(torch_dist, x_shape, event_dim, *args, **kwargs)
 
     def sample(self):
-        ps = self.torch_dist.probs.data
-        zero = ps.new(self._sample_shape + ps.shape).zero_()
-        indices = super(OneHotCategorical, self).sample().data
-        if indices.dim() < zero.dim():
-            indices = indices.unsqueeze(-1)
-        one_hot = zero.scatter_(-1, indices, 1)
-        return Variable(one_hot)
+        return self.torch_dist.sample(self._sample_shape)
 
     def batch_log_pdf(self, x):
```

Comment:
Delete this method?

Reply:
We are doing the masking and appending the (1,) dim so not sure if we can delete this.